src/app/+data-and-analytics/data-and-analytics.component.ts

```diff
@@ -326,7 +338,7 @@ export class DataAndAnalyticsComponent implements OnInit {
         return;
       }
 
-      if (!selectedIds.length) {
+      if (!selectedIds.length || selectedIds.length === (organization.people?.length ?? 0)) {
```

Comment:
If you are needing to do this (`selectedIds.length === (organization.people?.length ?? 0)`) here, then why not use the same logic above in the HTML where you have `(selectedPeople | async)?.length === organization.people.length`? (Or vice-versa.)

Example changing the HTML:
```js
[checked]="((selectedPeople | async)?.length ?? 0) === organization.people.length"
```

(And if you are ensuring 0, then the `===` might also be `==` for that matter.)

What is the expected logic here?
If `selectedIds.length` is 0 and `organization.people.length` is 0, then this is a match.
Is that intended?
Designated "selected" based on length is confusing me.
(I do see that the previous code does that as well.)